src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -2337,6 +2337,23 @@ public Type visitErrorType(ErrorType t, Symbol sym) {
         };
     // </editor-fold>
 
+    /** Determine whether the given outer class strictly encloses the inner one.
+     */
+    public boolean hasOuterClass(Type innerType, Type outerType) {
+        if (!innerType.hasTag(CLASS) || !outerType.hasTag(CLASS))
+            return false;
+        innerType = erasure(innerType);
```

Comment:
OK I'm glad you pointed that out because I'm a little unclear on the best way to do this bit.

Just to confirm, you are saying that this:

`if (erasure(type).equalsIgnoreMetadata(outerType)) {`

should be replaced with this?

`if (isSameType(type, outerType)) {`

Reply:
yes

Reply:
Thanks... updated in `6e96a7d76f8`.